### ISSUE TITLE
Cache issues on /news /lunchtime-lectures & the homepage

### ISSUE THREAD

**Issue by langphil** [theodi/shared#654]
Hi,

Just putting in an issue for caching issues on the website. Blog posts / news haven't updated since June 1. 

cc @Floppy @pezholio

**Comment by pezholio** [comment]
Have there been any published since then? The Content API isn't returning any different from the frontend, so if you're expecting to see anything else, the issue isn't caching.

**Comment by langphil** [comment]
Here is the main culprit - www.theodi.org/blog/invitation-to-tender-report-brief-generation-open

I’m about to publish a new Lunchtime Lecture also, so I’ll test with that. 

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:27:34, pezholio (notifications@github.com) wrote:

Have there been any published since then? The Content API isn't returning any different from the frontend, so if you're expecting to see anything else, the issue isn't caching.

**Comment by pezholio** [comment]
That one shows up on /news - Published date is 20th May

**Comment by langphil** [comment]
Ah!!

Now I feel silly. 

Apologies.

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:32:08, pezholio (notifications@github.com) wrote:

That one shows up on /news - Published date is 20th May

**Comment by langphil** [comment]
Will publish a lunchtime lecture now to test. 

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:32:49, Phil (phil.lang@theodi.org) wrote:

Ah!!

Now I feel silly. 

Apologies.

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:32:08, pezholio (notifications@github.com) wrote:

That one shows up on /news - Published date is 20th May

**Comment by langphil** [comment]
Hi, 

I’ve published a lecture > http://theodi.org/lunchtime-lectures/friday-lunchtime-lecture-open-data-powered-solutions-helping-people-get-better-jobs

It’s not generating on the page /lunchtime-lectures

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:33:21, Phil (phil.lang@theodi.org) wrote:

Will publish a lunchtime lecture now to test. 

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:32:49, Phil (phil.lang@theodi.org) wrote:

Ah!!

Now I feel silly. 

Apologies.

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 09:32:08, pezholio (notifications@github.com) wrote:

That one shows up on /news - Published date is 20th May

**Comment by pezholio** [comment]
Yep. There seems to be some sort of OTHER weird caching layer (or something) in Content API. I can't even.

**Comment by Floppy** [comment]
Yeesh. At least we know it less space to hide in now. Content API *shouldn't* be caching, but then... who can say. The old data is appearing in the API responses from content API in a separate browser, so it's definitely on that side?

**Comment by langphil** [comment]
There are similar issues happening with http://theodi.org/current-start-ups  to which I published 7 items to today. Could be related to the recent work that was done on these pages?

-- 
Phil Lang - Digital Production Editor
@ODIHQ | thodi.org
@langphil | 0741 44 63047

On 15 June 2015 at 12:12:21, James Smith (notifications@github.com) wrote:

Yeesh. At least we know it less space to hide in now. Content API shouldn't be caching, but then... who can say. The old data is appearing in the API responses from content API in a separate browser, so it's definitely on that side?

**Comment by Floppy** [comment]
Content API still has Rack::Cache in it, and does set expiry times, so it might be misbehaving in a similar way. Perhaps we can remove it from the middleware stack and see if that helps, as it did with frontend-www.

**Comment by pezholio** [comment]
I've rebooted and all seems to be well. I don't think we can fully relieve ourselves of Rack::Cache, as it's a dependency of `actionpack`, but we can certainly remove all the expires settings on the app, which should negate the problem.

**Comment by Floppy** [comment]
You can't bin it from the Gemfile, but you can remove it from the middleware stack - then you can keep the expiry headers to manage client-side caching. That's what we did on frontend-www.
